sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -53,47 +60,253 @@ def _is_ipython(self):
+            additional_metric_names: List of metrics that Katib collects from the
+                objective function in addition to objective metric.
+            objective_type: Type for the Experiment optimization for the objective metric.
+                Must be one of `minimize` or `maximize`.
+            objective_goal: Objective goal that Experiment should reach to be Succeeded.
+            max_trial_count: Maximum number of Trials to run.
+            parallel_trial_count: Number of Trials that Experiment runs in parallel.
+            max_failed_trial_count: Maximum number of Trials allowed to fail.
+            retain_trials: Whether Trials' resources (e.g. pods) are deleted after Succeeded state.
+            packages_to_install: List of Python packages to install before
+                executing the objective function.
+            pip_index_url: The PyPI url from which to install Python packages.
+        """
+
+        # Create Katib Experiment template.
+        experiment = models.V1beta1Experiment(
+            api_version=f"{constants.KUBEFLOW_GROUP}/{constants.KATIB_VERSION}",
+            kind=constants.EXPERIMENT_KIND,
+            metadata=models.V1ObjectMeta(name=name, namespace=namespace),
+            spec=models.V1beta1ExperimentSpec(),
+        )
+
+        # Add Objective to the Katib Experiment.
+        experiment.spec.objective = models.V1beta1ObjectiveSpec(
+            type=objective_type,
+            objective_metric_name=objective_metric_name,
+            additional_metric_names=additional_metric_names,
+        )
+        if objective_goal is not None:
+            experiment.spec.objective.goal = objective_goal
+
+        # Add Algorithm to the Katib Experiment.
+        experiment.spec.algorithm = models.V1beta1AlgorithmSpec(
+            algorithm_name=algorithm_name
+        )
+
+        # Add Trial budget to the Katib Experiment.
+        if max_trial_count is not None:
+            experiment.spec.max_trial_count = max_trial_count
+        if parallel_trial_count is not None:
+            experiment.spec.parallel_trial_count = parallel_trial_count
+        if max_failed_trial_count is not None:
+            experiment.spec.max_failed_trial_count = max_failed_trial_count
+
+        # Modify objective function.
+        # Check if objective function is callable.
+        if not callable(objective):
+            raise ValueError(
+                f"Objective function must be callable, got function type: {type(objective)}"
+            )
+
+        # Verify the objective function arguments.
+        objective_signature = inspect.signature(objective)
+        try:
+            objective_signature.bind({})
+        except Exception:
+            raise ValueError(
+                "Invalid args in the Objective function. "
+                "The function args must have only 'parameters' dictionary. "
+                f"Current Objective arguments: {objective_signature}"
+            ) from None
```

Comment:
How about extracting this part to a single function with naming something like `validate_objective_fucntion`?
I think this kinds of validation for user's wrong-use could become bigger and bigger.